### PR TITLE
planner/cascades: add a transformation rule FilterAggregateTranspose

### PR DESCRIPTION
**Pull request by zz-jason**
### What problem does this PR solve? 

Trying to add a new transformation rule, which pushes the filter through the aggregate.

### What is changed and how it works?

There is a little code change in the transformation framework. Group logical property has not derived.

test will be added later.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8665)

### CODE REVIEW

File: planner/cascades/rule_FilterAggregateTranspose.go [hunk 1]
```diff
@@ -0,0 +1,136 @@
+// Copyright 2018 PingCAP, Inc.
```
Comment (winoros) [inline planner/cascades/rule_FilterAggregateTranspose.go]: Seems that golang recommend file name only containing lowercase?

**Comment by winoros** [comment]
Codes inside the `onTransform` LGTM

File: planner/cascades/rule_FilterAggregateTranspose.go (hunk 1 above)
Comment (zz-jason) [reply planner/cascades/rule_FilterAggregateTranspose.go]: I'll refactor this later when UT is added.

**Comment by zz-jason** [comment]
It's hard to merge the master branch, I'll close this and file another PR.

**Pull request closed without merging**